### PR TITLE
Refactor with cargo-clippy and cargo-fmt

### PR DESCRIPTION
**Pull request by cyphersnake**
I've been reviewing your codebase and would like to suggest the adoption of Rust's clippy and fmt tools for this project.

Incorporating clippy could enhance code quality by catching common mistakes and potential bugs early. On the other hand, fmt can help enforce a consistent coding style for improved readability and maintenance. Both tools are highly regarded in the Rust community and would align this project more closely with ecosystem standards. 

I turned off a couple of rules that may not be very intuitive for non-Rust devs (clippy insists on iterating through iterators rather than indexes for example). If you have specific formatting preferences or don't like the suggested changes, I'd welcome any feedback. Almost all of the changes made here are `cargo fmt` + `cargo clippy --fix`.

If you decide that using Rust standards is appropriate for your project, these checks can be included at the github-actions level, which are free for open-source projects.

Thank you for considering this proposal!

### CODE REVIEW

File: src/bucket_msm.rs [hunk 1]
```diff
@@ -289,6 +324,7 @@ impl<P: Parameters> BucketMSM<P> {
                 }
                 total
             })
+            + lowest
```
Comment (drouyang) [inline src/bucket_msm.rs]: Moving "+ lowest" to the end does not make the code easier to read. It's easy to omit the last line. Why did we introduce this change?

File: src/glv.rs [hunk 2]
```diff
@@ -25,17 +25,23 @@ pub fn decompose_slow(scalar: &G1ScalarField) -> (G1ScalarField, G1ScalarField)
...
+    (
+        G1ScalarField::from(quotient),
+        G1ScalarField::from(remainder),
+    )
 }
 
 const LMDA1: u128 = 0xac45a4010001a402; // lambda high 64 bit
 const LMDA0: u128 = 0x00000000ffffffff; // lambda low 64 bit
-// const INV2: u128 = 0x1;                // 2**256 // lambda - 128th bit
+                                        // const INV2: u128 = 0x1;                // 2**256 // lambda - 128th bit
```
Comment (drouyang) [inline src/glv.rs]: We should revert this change. This line is supposed to start from the beginning

File: src/msm.rs [hunk 3]
```diff
@@ -74,89 +74,86 @@ impl VariableBaseMSM {
...
     }
 
-    pub fn multi_scalar_mul<P: Parameters>(points: &[GroupAffine<P>], scalars: &[BigInt<P>]) -> GroupProjective<P> {
+    pub fn multi_scalar_mul<P: Parameters>(
+        points: &[GroupAffine<P>],
+        scalars: &[BigInt<P>],
+    ) -> GroupProjective<P> {
         let opt_window_size = Self::get_opt_window_size(log2(points.len()));
-        Self::multi_scalar_mul_custom(&points, &scalars, opt_window_size, 2048, 256)
+        Self::multi_scalar_mul_custom(points, scalars, opt_window_size, 2048, 256)
```
Comment (drouyang) [inline src/msm.rs:153]: Why should the "&" be removed?

File: tests/bitmap_tests.rs [hunk 4]
```diff
@@ -3,30 +3,30 @@ mod bitmap_tests {
...
+        assert!(!bitmap.test_and_set(33));
     }
 
     #[test]
     fn test_set_when_more_than_once_return_true() {
         let mut bitmap = setup();
-        assert_eq!(bitmap.test_and_set(0), false);
-        assert_eq!(bitmap.test_and_set(0), true);
+        assert!(!bitmap.test_and_set(0));
+        assert!(bitmap.test_and_set(0));
```
Comment (drouyang) [inline tests/bitmap_tests.rs:20]: Changes here make the code shorter, but not necessarily easier to read. "true" and "false" looks good here.

**Review by drouyang — COMMENTED** [review]
This PR is too large to effectively review. In particular, it mixes pure format change with some syntax changes. I would suggest to isolate changes from cargo-fmt, so we can get those fmt changes merged first.

File: src/msm.rs (hunk 3 above)
Comment (cyphersnake) [reply src/msm.rs:153]: Since `points` & `scalaras` is already a reference, in fact, `&&` value was passed inside, which is pointless

File: tests/bitmap_tests.rs (hunk 4 above)
Comment (cyphersnake) [reply tests/bitmap_tests.rs:20]: Comparing `true` and `false` via `assert_eq` is bad form. There are no manual changes here, all code changes are made in accordance with the cargo-clippy rules. 

However, here for better readability, we can add `assert_false!` macro (or use https://docs.rs/all_asserts/latest/all_asserts/macro.assert_false.html). What do you think?

File: src/msm.rs (hunk 3 above)
Comment (drouyang) [reply src/msm.rs:153]: Makes sense

File: tests/bitmap_tests.rs (hunk 4 above)
Comment (drouyang) [reply tests/bitmap_tests.rs:20]: `assert_false!` sounds good to me

**Comment by cyphersnake** [comment]
Will be split into two separate PRs
